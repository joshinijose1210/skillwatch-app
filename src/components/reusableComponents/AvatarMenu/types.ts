export interface AvatarMenuProps {
    open: boolean;
    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
}
