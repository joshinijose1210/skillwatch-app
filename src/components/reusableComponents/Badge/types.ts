export type Props = {
    children?: React.ReactNode;
    bgColor?: string;
    pd?: string;
    borderRadius?: string;
};
