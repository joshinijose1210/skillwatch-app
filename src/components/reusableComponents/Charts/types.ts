import { ReactNode } from 'react';

export interface LineChartProps {
    rating: number;
    ratingBy?: string;
}

export interface BarChartProps {
    rating: number;
}

export type DataChunk = {
    id: string;
    label: string;
    value: number;
    color: string;
};

export type PieChartPropTypes = {
    data: DataChunk[];
    title: ReactNode;
    noData?: boolean;
    padAngle?: number;
    enableArcLabels?: boolean;
    enableArcLinkLabels?: boolean;
    margin?: number;
    isInteractive?: boolean;
};
