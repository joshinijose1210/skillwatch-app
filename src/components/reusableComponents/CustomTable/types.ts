import { RowHoverActionsType, TableColumnConfig } from '@medly-components/core';
import { SortOrder } from '@medly-components/core/dist/es/components/Table/types';

interface CustomTableBaseProps {
    children?: React.ReactNode;
    columns: TableColumnConfig[];
    count: number;
    data: any;
    isLoading: boolean;
    activePage: number;
    rowAction?: RowHoverActionsType;
    defaultSortField?: string;
    defaultSortOrder?: SortOrder;
    tableKey?: string;
    itemsPerPage?: number;
    setSortOrder?: (order: string) => void;
    withMinimap?: boolean;
}

interface WithPaginationProps extends CustomTableBaseProps {
    withPagination: true;
    handlePageChange: (page: number) => void;
}

interface WithoutPaginationProps extends CustomTableBaseProps {
    withPagination?: false;
    handlePageChange?: (page: number) => void;
}

// Union type
export type CustomTableProps = WithPaginationProps | WithoutPaginationProps;
