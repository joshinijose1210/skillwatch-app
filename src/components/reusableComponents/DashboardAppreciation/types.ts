export interface Appreciation {
    isExternalFeedback?: boolean;
    externalFeedbackFromEmailId?: string;
    appreciationToEmployeeId: string;
    appreciationToFirstName: string;
    appreciationFromId: string;
    appreciationToId: string;
    appreciationToLastName: string;
    appreciation: string;
    appreciationFromEmployeeId: string;
    appreciationFromRoleName: string;
    appreciationFromFirstName: string;
    appreciationFromLastName: string;
    appreciationToRoleName: string;
    submitDate: Date;
    isDummy?: boolean;
}

export type DashboardAppreciationPropType = {
    reviewCycleId: number;
};
