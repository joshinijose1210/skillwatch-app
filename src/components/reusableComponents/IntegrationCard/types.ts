import { ReactNode } from 'react';

export interface Props {
    icon: ReactNode;
    title?: string;
    description?: string;
    isComingSoon?: boolean;
    footerText?: string;
    content?: ReactNode;
    footer?: ReactNode;
}
