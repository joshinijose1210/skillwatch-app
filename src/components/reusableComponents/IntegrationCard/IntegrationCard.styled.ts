import { Card } from '@medly-components/core';
import styled from 'styled-components';

export const CardContainer = styled(Card)`
    width: 28rem;
    height: fit-content;
    display: flex;
    flex: 0 0 25%;
    justify-content: flex-start;
    position: relative;
    transition: box-shadow 0.3s;
    border-radius: 1rem;
    /* this component is probably not being used anywhere, might be removed later */
    border: 0.1rem solid ${({ theme }) => theme.colors.yellow[400]};
    background: ${({ theme }) => theme.colors.white};
    :hover {
        box-shadow: 0 0 1.1rem rgba(33, 33, 33, 0.4);
    }
`;

export const StyledLogoSection = styled('div')`
    background-color: ${({ theme }) => theme.colors.white};
    width: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 4rem 0;
`;

export const ComingSoonCard = styled(Card)`
    position: absolute;
    backdrop-filter: blur(0.5rem);
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.7);
    color: ${({ theme }) => theme.colors.white};
    display: flex;
    justify-content: center;
    align-items: center;
    opacity: 0;
    transition: opacity 0.25s;

    > * {
        transform: translateY(2rem);
        transition: transform 0.25s;
    }

    :hover {
        opacity: 1;
    }

    :hover > * {
        transform: translateY(0);
    }
`;

export const ContentSection = styled('div')`
    display: flex;
    flex-direction: column;
    gap: 1rem;
    align-items: center;
    padding: 2rem 1rem 0 1rem;
    /* this component is probably not being used anywhere, might be removed later */
    background-color: ${({ theme }) => theme.colors.blue['400']};
`;

export const FooterSection = styled('div')`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 2rem 0;
    width: 100%;
    /* this component is probably not being used anywhere, might be removed later */
    background-color: ${({ theme }) => theme.colors.blue['400']};
`;
