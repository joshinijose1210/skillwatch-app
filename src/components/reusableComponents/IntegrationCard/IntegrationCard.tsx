import { Text } from '@medly-components/core';
import { defaultTheme } from '@theme';
import { FC } from 'react';
import { CardContainer, ComingSoonCard, ContentSection, FooterSection, StyledLogoSection } from './IntegrationCard.styled';
import { Props } from './types';

export const IntegrationCard: FC<Props> = ({ icon, title, description, content, footer, footerText, isComingSoon }) => {
    return (
        <CardContainer withoutPadding={true}>
            <StyledLogoSection>{icon}</StyledLogoSection>
            {(title || description || content) && (
                <ContentSection>
                    {title && (
                        <>
                            <Text textVariant="h3" textAlign="center" textWeight="Medium">
                                {title}
                            </Text>
                        </>
                    )}

                    {description && (
                        <>
                            <Text textVariant="body1" textAlign="center" textWeight="Regular">
                                {description}
                            </Text>
                        </>
                    )}

                    {content && <>{content}</>}
                </ContentSection>
            )}
            <FooterSection>
                {footer && <>{footer}</>}
                {footerText && (
                    <>
                        <Text textColor={defaultTheme.colors.blue[800]} textWeight="Strong" textVariant="h4">
                            {footerText}
                        </Text>
                    </>
                )}
            </FooterSection>
            {isComingSoon && (
                <ComingSoonCard>
                    <Text textAlign="center" textVariant="h2">
                        Coming soon...
                    </Text>
                </ComingSoonCard>
            )}
        </CardContainer>
    );
};
