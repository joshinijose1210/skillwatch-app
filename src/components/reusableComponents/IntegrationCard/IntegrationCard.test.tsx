import { Text } from '@medly-components/core';
import { renderWithRouter, screen } from '@utils/test-utils';
import { IntegrationCard } from './IntegrationCard';
import { Props } from './types';

const defaultProps: Props = {
        icon: <svg />,
        title: 'Dummy title'
    },
    dashboardCardRender = (props = defaultProps) => renderWithRouter(<IntegrationCard {...props} />);

describe('IntegrationCard', () => {
    it('should render properly with default props', () => {
        const { container } = dashboardCardRender();
        expect(screen.getByText('Dummy title')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });

    it('should render properly with all props given', () => {
        const { container } = dashboardCardRender({
            ...defaultProps,
            description: 'dummy Description',
            content: <Text>dummy content</Text>,
            footer: <Text>footer content</Text>,
            footerText: 'dummy footer text'
        });
        expect(screen.getByText('dummy Description')).toBeInTheDocument();
        expect(screen.getByText('dummy content')).toBeInTheDocument();
        expect(screen.getByText('footer content')).toBeInTheDocument();
        expect(screen.getByText('dummy footer text')).toBeInTheDocument();
        expect(container).toMatchSnapshot();
    });
});
