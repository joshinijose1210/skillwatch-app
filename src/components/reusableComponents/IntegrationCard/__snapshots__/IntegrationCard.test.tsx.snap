// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntegrationCard should render properly with all props given 1`] = `
<div>
  <div
    alignitems="center"
    backgroundcolor="#ffffff"
    borderradius="0.8rem"
    class="sc-gr411i-0 dxPiGG sc-9frze1-0 cWPDUF"
    display="block"
    flex="1"
    flowdirection="vertical"
    padding="[object Object]"
    separatorcolor="#98A7B7"
    shadowcolor="#b0bcc8"
    variant="solid"
  >
    <div
      class="sc-9frze1-1 bEplZS"
    >
      <svg />
    </div>
    <div
      class="sc-9frze1-3 iMElqA"
    >
      <h3
        class="sc-1ntgl3l-0 OLixH"
        textalign="center"
        textcolor="inherit"
        textvariant="h3"
        textweight="Medium"
      >
        Dummy title
      </h3>
      <span
        class="sc-1ntgl3l-0 hSbYjV"
        textalign="center"
        textcolor="inherit"
        textvariant="body1"
        textweight="Regular"
      >
        dummy Description
      </span>
      <span
        class="sc-1ntgl3l-0 hGagSU"
        textalign="initial"
        textcolor="inherit"
      >
        dummy content
      </span>
    </div>
    <div
      class="sc-9frze1-4 fAghlw"
    >
      <span
        class="sc-1ntgl3l-0 hGagSU"
        textalign="initial"
        textcolor="inherit"
      >
        footer content
      </span>
      <h4
        class="sc-1ntgl3l-0 fmUXqp"
        textalign="initial"
        textcolor="#1C3969"
        textvariant="h4"
        textweight="Strong"
      >
        dummy footer text
      </h4>
    </div>
  </div>
</div>
`;

exports[`IntegrationCard should render properly with default props 1`] = `
<div>
  <div
    alignitems="center"
    backgroundcolor="#ffffff"
    borderradius="0.8rem"
    class="sc-gr411i-0 dxPiGG sc-9frze1-0 cWPDUF"
    display="block"
    flex="1"
    flowdirection="vertical"
    padding="[object Object]"
    separatorcolor="#98A7B7"
    shadowcolor="#b0bcc8"
    variant="solid"
  >
    <div
      class="sc-9frze1-1 bEplZS"
    >
      <svg />
    </div>
    <div
      class="sc-9frze1-3 iMElqA"
    >
      <h3
        class="sc-1ntgl3l-0 OLixH"
        textalign="center"
        textcolor="inherit"
        textvariant="h3"
        textweight="Medium"
      >
        Dummy title
      </h3>
    </div>
    <div
      class="sc-9frze1-4 fAghlw"
    />
  </div>
</div>
`;
